docs/configuring-kubelinter.md

```diff
@@ -142,3 +148,64 @@ For example,
         key: company.io/responsible
       remediation: Please set the annotation 'company.io/responsible'. This will be parsed by xy to generate some docs.
   ```
+
+### Custom `objectKinds`
+
+If you want to add a check for an objectKind that isn't built into kube-linter, you can also register your own custom objectKind alongside your custom check. This is especially useful for resources from CRDs, where the objectKind may not exist in every cluster, and isn't a good candidate for upstream support.
+
+Below is an example of a check that looks for excessively long certificate lifetimes in CNCF `cert-manager` Certificate resources.
+
+- First, define a constant that can be used to reference the objectKind in your check
+
+  ```go
+  const (
+    Certificate = cmv1.CertificateKind
+  )
+  ```
+
+- Next, define the GVK of the object that we want to match
+
+  ```go
+  var (
+    certificateGVK = cmv1.SchemeGroupVersion.WithKind(Certificate)
+  )
+  ```
+
+- Now we can register the matcher function. `RegisterObjectKind()` adds your objectKind and its corresponding `MatcherFunc` to the global registry so that the linter knows about it and can check input objects for a match.
+
+  ```go
+  func init() {
+    // This function can contain any arbitrary logic to match objects we want to check with this lint
+    objectkinds.RegisterObjectKind(Certificate, objectkinds.MatcherFunc(func(gvk schema.GroupVersionKind) bool {
+      return gvk == certificateGVK
+    }))
+  }
+  ```
+
+- Finally, the check template itself
+
+  ```go
```

Comment:
I wonder if we could add this as [testable example](https://go.dev/blog/examples) to get fast feedback if we broke something and need to update docs.

Reply:
Cool, T had no idea that was a feature. I've rewritten the doc into a go file with the original code plus a quick test, and changed the markdown to refer to the go file. The example will be tested on `make test`, so we can be notified of any regressions.